File: tests/de/umr/fixcon/wrappers/Problem_Test.java
package de.umr.fixcon.wrappers;

import com.google.common.graph.GraphBuilder;
import de.umr.fixcon.graphFunctions.standardFunctions.EdgeCount;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Problem_Test {

    @Test
    void constructor_test() {
        Problem p = new Problem(GraphBuilder.undirected().build(), 5, new EdgeCount());
        assertEquals(GraphBuilder.undirected().build(), p.graph);
        assertEquals(5, p.subgraphSize);
        assertEquals(0, p.parameters.length);

        p = new Problem(GraphBuilder.undirected().build(), 3, new EdgeCount(), 1, 2, 3, 4, 5);
        assertEquals(GraphBuilder.undirected().build(), p.graph);
        assertEquals(3, p.subgraphSize);
        assertEquals(5, p.parameters.length);
    }
}

File: tests/de/umr/fixcon/wrappers/Problem_Test.kt
package de.umr.fixcon.wrappers

import de.umr.core.dataStructures.VertexOrderedGraph
import de.umr.fixcon.Problem
import de.umr.fixcon.graphFunctions.EdgeCountFunction
import org.jgrapht.Graphs.addEdgeWithVertices
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class Problem_Test {
    @Test
    fun constructor_test() {
        val g = VertexOrderedGraph<Int>()
        addEdgeWithVertices(g, 1, 2)
        var p = Problem(g, EdgeCountFunction(5))

        assertEquals(2, p.g.vertexSet().size)
        assertEquals(5, p.function.k)

        p = Problem(g, EdgeCountFunction(3))
        assertEquals(3, p.function.k)
    }
}